src/main/java/core/dao/car/CarDaoJdbc.java

```diff
@@ -0,0 +1,215 @@
+            insertCarStatement.executeUpdate();
+            ResultSet resultSet = insertCarStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject("id", Long.class));
+            }
+            insertDrivers(car, connection);
+        } catch (SQLException e) {
+            throw new DataProcessingException(String
+                    .format(ErrorMessages.ADD.getMessage(),
+                            Car.class.getSimpleName(), car), e);
+        }
+        
+        return car;
+    }
+    
+    @Override
+    public Optional<Car> get(Long id) {
+        String select = "SELECT cars.id as car_id, cars.model as car_model,"
+                        + " cars.manufacturer as manufacturer_id,"
+                        + " m.name as manufacturer_name, m.country as manufacturer_country,"
+                        + " d.id as driver_id, d.name as driver_name,"
+                        + " d.licence_number as driver_licence"
+                        + " FROM cars"
+                        + " LEFT JOIN manufacturers m ON m.id = cars.manufacturer"
+                        + " LEFT JOIN cars_drivers cd ON cd.\"car_Id\" = cars.id"
+                        + " LEFT JOIN drivers d on cd.\"driver_Id\" = d.id"
+                        + " WHERE cars.id = ? AND cars.deleted = false "
+                        + "AND (d.deleted = false OR d.deleted IS NULL)";
+        Car car = null;
+        try (Connection connection = ConnectionUtils.getConnection();
+                PreparedStatement getByIdStatement = connection.prepareStatement(select)) {
+            getByIdStatement.setLong(1, id);
+            ResultSet resultSet = getByIdStatement.executeQuery();
+            if (resultSet.next()) {
+                car = DaoUtils.parseToCar(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(String
+                    .format(ErrorMessages.GET.getMessage(),
+                            Car.class.getSimpleName(), id), e);
+        }
+        return Optional.ofNullable(car);
+    }
+    
+    @Override
+    public List<Car> getAll() {
+        List<Car> returnList = new ArrayList<>();
+        String getAll = "SELECT cars.id as car_id, cars.model as car_model,"
+                        + " cars.manufacturer as manufacturer_id,"
+                        + " m.name as manufacturer_name, m.country as manufacturer_country,"
+                        + " d.id as driver_id, d.name as driver_name,"
+                        + " d.licence_number as driver_licence"
+                        + " FROM cars"
+                        + " JOIN manufacturers m ON m.id = cars.manufacturer"
+                        + " JOIN cars_drivers cd ON cd.\"car_Id\" = cars.id"
+                        + " JOIN drivers d on cd.\"driver_Id\" = d.id"
+                        + " WHERE cars.deleted = false AND (d.deleted = false OR d.deleted IS NULL)"
```

Comment:
same here and in `getAllByDriverId()` methods - better use two queries

Reply:
Once again, I had the two queries solution which shows 10 times lower performance than this one. I would like to stay with a single query.